stock_request/models/stock_request.py

```diff
@@ -362,6 +394,21 @@ def _action_launch_procurement_rule(self):
             if float_compare(qty, request.product_qty, precision_digits=precision) >= 0:
                 continue
 
+            # If stock is available we use it and we do not execute rule
+            if request.company_id.stock_request_check_available_first:
+                quants = quant_model._gather(request.product_id, request.location_id)
```

Comment:
This is still not valid, as you have to make as many moves as possible locations. That's why I told you about first getting qty_available from product with the proper context, and then do the moves with `_gather`.

Reply:
Changes done to take into account quants and create the necessary moves.